packages/trader/src/Modules/Trading/Components/Form/DatePicker/trading-date-picker.tsx

```diff
@@ -25,14 +37,15 @@ const TradingDatePicker = observer(({ id, is_24_hours_contract, mode, name }) =>
 
     const isMounted = useIsMounted();
 
-    const [disabled_days, setDisabledDays] = React.useState([]);
-    const [market_events, setMarketEvents] = React.useState([]);
+    const [disabled_days, setDisabledDays] = React.useState<number[]>([]);
+    const [market_events, setMarketEvents] = React.useState<TMarketEvent[]>([]);
     const [duration, setDuration] = React.useState(current_duration);
-    const [selected_date, setSelectedDate] = React.useState();
+    const [selected_date, setSelectedDate] = React.useState<moment.Moment>();
 
     React.useEffect(() => {
         onChangeCalendarMonth();
-    }, [onChangeCalendarMonth]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
adding onChangeCalendarMonth to the array of dependencies requires onChangeCalendarMonth to be declared before useEffect which will cause infinite loop. onChangeCalendarMonth callback should be called once on the component render, and then it's passed to and called from its children.